### PR TITLE
Add an RRF Retriever

### PR DESCRIPTION
**Pull request by jdconrad**
This change adds an `RRFRetrieverBuilder` using the same strategy as `ClassicRetrieverBuilder` where we extract the parsed request body into a `SearchSourceBuilder`.

### CODE REVIEW

File: x-pack/plugin/rank-rrf/src/main/java/org/elasticsearch/xpack/rank/rrf/RRFRetrieverBuilder.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    public static final ParseField WINDOW_SIZE_FIELD = new ParseField("window_size");
+    public static final ParseField RANK_CONSTANT_FIELD = new ParseField("rank_constant");
+
+    public static final ObjectParser<RRFRetrieverBuilder, RetrieverParserContext> PARSER = new ObjectParser<>(
+        RRFRankPlugin.NAME,
+        RRFRetrieverBuilder::new
+    );
+
+    static {
+        PARSER.declareObjectArray((v, l) -> v.retrieverBuilders = l, (p, c) -> {
```
Comment (mayya-sharipova) [inline x-pack/plugin/rank-rrf/src/main/java/org/elasticsearch/xpack/rank/rrf/RRFRetrieverBuilder.java:38]: This looks like we are declaring "retrievers" field as optional for the parser, should we make it required?

**Review by mayya-sharipova — APPROVED** [review]
@jdconrad Thanks! Overall LGTM, I left a comment about making "retrievers" field required.

File: x-pack/plugin/rank-rrf/src/main/java/org/elasticsearch/xpack/rank/rrf/RRFRetrieverBuilder.java (hunk 1 above)
Comment (jdconrad) [reply x-pack/plugin/rank-rrf/src/main/java/org/elasticsearch/xpack/rank/rrf/RRFRetrieverBuilder.java:38]: I agree with you on the required. This should probably be a constructing object parser as well like you did for knn. I will make these changes in a follow up. For now this is protected by the fact we extract to sub_searches and you need at least 2 queries there. However, there error message would be really confusing to get.

Comment (jdconrad) [reply x-pack/plugin/rank-rrf/src/main/java/org/elasticsearch/xpack/rank/rrf/RRFRetrieverBuilder.java:38]: I agree with you on the required. This should probably be a constructing object parser as well like you did for knn. I will make these changes in a follow up. For now this is protected by the fact we extract to sub_searches and you need at least 2 queries there. However, there error message would be really confusing to get.

**Comment by jdconrad** [comment]
Thank you for the review, @mayya-sharipova!

**Pull request merged**